pyband/client.py

```diff
@@ -1,213 +1,120 @@
+                account_any.Unpack(account)
+                return account
+        except:
+            return None
 
-    def get_request_id_by_tx_hash(self, tx_hash: HexBytes) -> List[int]:
-        msgs = self._get("/txs/{}".format(tx_hash.hex()))["logs"]
+    # return list of request, if no request, get from report
+    def get_request_id_by_tx_hash(self, tx_hash: bytes) -> List[str]:
+        tx = self.stubTx.GetTx(tx_service.GetTxRequest(hash=tx_hash))
         request_ids = []
-        for msg in msgs:
-            request_event = [event for event in msg["events"] if event["type"] == "request"]
+        for tx in tx.tx_response.logs:
+            request_event = [event for event in tx.events if event.type == "request" or event.type == "report"]
             if len(request_event) == 1:
-                attrs = request_event[0]["attributes"]
-                attr_id = [attr for attr in attrs if attr["key"] == "id"]
+                attrs = request_event[0].attributes
+                attr_id = [attr for attr in attrs if attr.key == "id"]
                 if len(attr_id) == 1:
-                    request_id = attr_id[0]["value"]
-                    request_ids.append(int(request_id))
+                    request_id = attr_id[0].value
+                    request_ids.append(request_id)
         if len(request_ids) == 0:
-            raise EmptyRequestMsgError("There is no request message in this tx")
+            raise NotFoundError("Request Id is not found")
         return request_ids
+                     
+    def send_tx_sync_mode(self, tx_byte: bytes) -> abci_type.TxResponse:
+        return self.stubTx.BroadcastTx(
+            tx_service.BroadcastTxRequest(tx_bytes=tx_byte, mode=tx_service.BroadcastMode.BROADCAST_MODE_SYNC)
+        ).tx_response
+
+    def send_tx_async_mode(self, tx_byte: bytes) -> abci_type.TxResponse:
+        return self.stubTx.BroadcastTx(
+            tx_service.BroadcastTxRequest(tx_bytes=tx_byte, mode=tx_service.BroadcastMode.BROADCAST_MODE_ASYNC)
+        ).tx_response
+
+    def send_tx_block_mode(self, tx_byte: bytes) -> abci_type.TxResponse:
+        return self.stubTx.BroadcastTx(
+            tx_service.BroadcastTxRequest(tx_bytes=tx_byte, mode=tx_service.BroadcastMode.BROADCAST_MODE_BLOCK)
+        ).tx_response
 
-    def get_reference_data(self, pairs: List[str], min_count: int, ask_count: int) -> List[ReferencePrice]:
-        symbols = set([symbol for pair in pairs for symbol in pair.split("/") if symbol != "USD"])
-        data = self._post(
-            "/oracle/request_prices",
-            json={
-                "symbols": list(symbols),
-                "min_count": min_count,
-                "ask_count": ask_count,
-            },
+    def get_chain_id(self) -> str:
+        latest_block = self.get_latest_block()
+        return latest_block.block.header.chain_id
+
+    def get_reference_data(self, pairs: [str], min_count: int, ask_count: int) -> oracle_query.QueryRequestPriceResponse:
```

Comment:
1. pairs: List[str]
2. This function need to convert pair and calculate reference price same as the old version do, so return type should be the same on old one